packages/compass-indexes/src/components/search-indexes-modals/base-search-index-modal.tsx

```diff
@@ -249,6 +319,36 @@ export const BaseSearchIndexModal: React.FunctionComponent<
                 />
               </section>
               <HorizontalRule />
+              {isVectorSearchSupported && (
```

Comment:
nit: I like the idea of keeping the `searchIndexTypes` list instead of just hardcoding two options as radio boxes here, but then the logical next step for me would be to also decouple the checks for the absense/presence of individual types from the React render tree here, e.g. filter `searchIndexTypes` at the top of the component and then just do something along the lines of 
```suggestion
              {filteredSearchIndexTypes.length > 1 && (
```

here

Reply:
If it's alright with you I'd prefer to hold off on that refactor until theres more complexity to handle here, like more atlas search index types.